### PR TITLE
Finding an account for contacts/opportunities by entering

### PR DESCRIPTION
**Pull request by unijewski**
https://github.com/fatfreecrm/fat_free_crm/issues/397

I followed your code convention but it sometimes doesn't cover with ruby style guide. I'm going to do code refactoring based on `rails4` branch. I'll create a new pull request with it.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://unijewski-issue-397.fatfreecrm-fat-free-crm-1bbda9a74e1b4e1e2ffd.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://unijewski-issue-397.fatfreecrm-fat-free-crm-1bbda9a74e1b4e1e2ffd.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/models/entities/account.rb [hunk 1]
```diff
@@ -113,8 +113,12 @@ def discard!(attachment)
   # Class methods.
   #----------------------------------------------------------------------------
   def self.create_or_select_for(model, params)
+    account_by_name = Account.find_by(name: params[:name])
+
     if params[:id].present?
       account = Account.find(params[:id])
+    elsif account_by_name.present?
+      account = account_by_name
```
Comment (johnnyshields) [inline app/models/entities/account.rb]: Consider doing this as the following:

```
    elsif params[:name].present?
      account = Account.find_by(name: params[:name])
```

Reason being that you will execute a query when you run this line:
```
account_by_name = Account.find_by(name: params[:name])
```

Comment (unijewski) [reply app/models/entities/account.rb]: Good catch, after such time I don't know why I implemented it in this way.

**Comment by johnnyshields** [comment]
@CloCkWeRX can we re-trigger the tests on this then merge? Looks good to me.

**Comment by CloCkWeRX** [comment]
Lets redo this one against more recent master

**Comment by johnnyshields** [comment]
Replaces #678

**Pull request closed without merging**